### PR TITLE
Development

### PR DESCRIPTION
**Pull request by hinaldobariya**
Category : Add, FindAll, FindById, Delete, Update, and Working on FindByTitle

### CODE REVIEW

File: ExpressNode/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+node_modules
+.idea
```
Comment (suresh-jarvis) [inline ExpressNode/.gitignore]: Press Enter after EOF

File: ExpressNode/ExpressRoute/employee.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  {name: 'reena', id: 104}
+]
+
+exports.findAll = function (req, res) {
+  res.send(employee)
+};
+
+exports.findById = function(req,res) {
+  res.send({id:req.params.id, name:'name',description:'description'})
+};
```
Comment (suresh-jarvis) [inline ExpressNode/ExpressRoute/employee.js]: Press Enter after EOF

File: ExpressNode/server.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+employee = require('./ExpressRoute/employee');
+
+var app = express();
+
+app.get('/employee', employee.findAll);
+app.get('/employee/:id', employee.findById);
+
+app.listen(8000);
+
+console.log("Running port 8000..");
```
Comment (suresh-jarvis) [inline ExpressNode/server.js]: Press Enter after EOF

File: ExpressNode/server.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+var express = require('express'),
```
Comment (suresh-jarvis) [inline ExpressNode/server.js]: Use const only

File: src/action/categoryAction.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
+const categoryModel = require('../model/category');
+
+module.exports = {
+
+  addCategory: function (req, res) {
+    const {title} = req.body;
+    const categoryData = {title};
+    categoryModel.insert(categoryData).then((data) => {
+      res.json({status: true, data: data, message: 'Add Success...'})
```
Comment (suresh-jarvis) [inline src/action/categoryAction.js]: If you have same key then no need to define key: value

eg : `res.json({status: true, data, message: 'Add Success.'})`

Comment (suresh-jarvis) [reply src/action/categoryAction.js]: Use proper message like `Category added successfully.`

File: src/action/categoryAction.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  getById: function(req,res) {
+    const {id} = req.params;
+    categoryModel.findById(id).then((data) => {
+      res.json({status:true, data:data})
+    })
+  },
+
+  getByTitle : function (req, res) {
+    const { title } = req.params;
+    console.log("title...",title);
```
Comment (suresh-jarvis) [inline src/action/categoryAction.js]: Dont put console.log in commit

File: src/action/categoryAction.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    categoryModel.findByTitle(title).then((data) => {
+      res.json({status: true, data:data})
+    })
+  },
+
+  delete: function(req,res) {
+    const {id} = req.params;
+    console.log("Id-------",id);
```
Comment (suresh-jarvis) [inline src/action/categoryAction.js]: Remove it

File: src/action/categoryAction.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    categoryModel.delete(id).then(() => {
+      res.json({status:true, message:'deleted...'})
```
Comment (suresh-jarvis) [inline src/action/categoryAction.js]: Proper Message please

File: src/model/category.js [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+  //isActive: Boolean
+});
+
+const Category = mongoose.model('category', CategorySchema);
+
+module.exports = {
+
+  insert: function (data) {
+    return new Promise((resolve) => {
+      data['createdAt'] = moment().format('YYYY-MM-DD');
```
Comment (suresh-jarvis) [inline src/model/category.js]: Dont define format in createdAt field that's require time also.
Like this : `data['createdAt'] = moment();`

File: src/model/category.js [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+  update: function (id, data) {
+    return new Promise((resolve) => {
+      this.findById(id).then((resData) => {
+        if (Object.keys(resData).length > 0) {
+
+          Object.keys(data).map(rKey => {
+            resData[rKey] = data[rKey];
+          });
+
+          resData['updatedAt'] = moment().format('YYYY-MM-DD');
```
Comment (suresh-jarvis) [inline src/model/category.js]: Dont use format

File: src/model/db.js [hunk 11]
```diff
@@ -0,0 +1,15 @@
+const mongoose = require('mongoose');
+
+mongoose.connect('mongodb://localhost/demo');
```
Comment (suresh-jarvis) [inline src/model/db.js]: Create config.js file for DB credential and connection string

File: src/model/user.js [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+    return new Promise((resolve) => {
+      User.findOne({email}, function (err, data) {
+        resolve((err) ? {} : (data == null) ? {} : data);
+      })
+    })
+  },
+
+  findByCredential: function (email, password) {
+    return new Promise((resolve) => {
+      console.log('Email---------',email);
```
Comment (suresh-jarvis) [inline src/model/user.js]: remove it from all files before commit

File: src/route/categoryRoute.js [hunk 13]
```diff
@@ -0,0 +1,12 @@
+const express = require('express');
+const router = express.Router();
+const categoryAction = require('../action/categoryAction');
+
+router.post('/',categoryAction.addCategory);
+router.get('/',categoryAction.getAllCategory);
+//router.get('/:id',categoryAction.getById);
+router.get('/:title',categoryAction.getByTitle);
```
Comment (suresh-jarvis) [inline src/route/categoryRoute.js]: why create route with /:title ?
Use only :id

File: ExpressNode/.gitignore (hunk 1 above)
Comment (hinaldobariya) [reply ExpressNode/.gitignore]: Ok , I will do

File: src/action/categoryAction.js (hunk 5 above)
Comment (hinaldobariya) [reply src/action/categoryAction.js]: Message modified.

File: src/model/category.js (hunk 9 above)
Comment (hinaldobariya) [reply src/model/category.js]: Ok, Remove moment() format.

File: src/model/db.js (hunk 11 above)
Comment (hinaldobariya) [reply src/model/db.js]: ok, I Created config.js File

File: src/route/categoryRoute.js (hunk 13 above)
Comment (hinaldobariya) [reply src/route/categoryRoute.js]: Yess I'm working on this part right now and now I'm converting from params to queryString Also

Comment (KanjariyaHardik) [reply src/route/categoryRoute.js]: Please commit your changes and push them

**Pull request closed without merging**